### ISSUE TITLE
Reset support needed in H5P-plugin

### ISSUE THREAD

**Issue by KSLaring** [h5p/h5p-moodle-plugin#150]
When developing courses or when reusing exisiting courses in Moodle, it is common to reset the course to clear all user data, e.g. prior attempts and so on.

Course reset is currently not supported by the H5P-activity, and we get a "Reset not supported" error message when we reset a course, which leaves us in doubt about how to proceed. Is user data still in there and we really need to delete the H5P-activity and recreate it? Is user data dropped? If teachers are unsure about how the module works, they might choose not to use it.

In short, support for course reset is needed.

**Comment by thomasmars** [comment]
Hi, H5P only stores user data if you have enabled "save content state".
I have created an [issue in the public tracker](https://h5ptechnology.atlassian.net/browse/HFP-1040) to make sure you can reset this when resetting the rest of the course.
Other data is stored within Moodle's APIs (e.g. gradebook), which can already be reset.

**Comment by foxlapinou** [comment]
Hi,
It's sad to see that more than 2 years later, reset functionnality still hasn't been included.
I hope that, better integration of H5P into Moodle being something planned to Moodle 3.8 (see https://docs.moodle.org/dev/Roadmap#Moodle_3.8_and_beyond:_in_progress), this will come in soon.

**Comment by otacke** [comment]
@foxlapinou H5P is an open source project, and the small H5P core team can't do it all on their own. Just because someone utters a wish for a feature, they can't promise to implement it (in a timely manner). So, how could more people be motivated to contribute code to the development or fund it?
